gradoop-flink/src/main/java/org/gradoop/common/config/GradoopStoreConfig.java

```diff
@@ -16,42 +16,34 @@
 package org.gradoop.common.config;
 
 import org.apache.flink.api.java.ExecutionEnvironment;
-import org.gradoop.common.model.api.entities.EPGMEdge;
-import org.gradoop.common.model.api.entities.EPGMGraphHead;
-import org.gradoop.common.model.api.entities.EPGMVertex;
-import org.gradoop.common.storage.api.PersistentEdgeFactory;
-import org.gradoop.common.storage.api.PersistentGraphHeadFactory;
-import org.gradoop.common.storage.api.PersistentVertexFactory;
+import org.apache.flink.util.Preconditions;
 import org.gradoop.flink.model.api.layouts.GraphCollectionLayoutFactory;
 import org.gradoop.flink.model.api.layouts.LogicalGraphLayoutFactory;
 import org.gradoop.flink.model.impl.layouts.gve.GVECollectionLayoutFactory;
 import org.gradoop.flink.model.impl.layouts.gve.GVEGraphLayoutFactory;
 import org.gradoop.flink.util.GradoopFlinkConfig;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
 /**
  * Basic configuration for Gradoop Stores
  *
- * @param <G> EPGM graph head type
- * @param <V> EPGM vertex type
- * @param <E> EPGM edge type
+ * @param <GF> graph head factory
+ * @param <VF> vertex factory
+ * @param <EF> edge factory
  */
-public abstract class GradoopStoreConfig<G extends EPGMGraphHead, V extends EPGMVertex, E extends EPGMEdge>
-  extends GradoopFlinkConfig {
+public abstract class GradoopStoreConfig<GF, VF, EF> extends GradoopFlinkConfig {
```

Comment:
hm...shouldn't GF, VGF and EF extend the Factories to ensure that factory classes are taken at this point? e.g. if someone implements its own GradoopStoreConfig?

Reply:
I thought this before, but I found this

```
PersistentGraphHeadFactory<G extends EPGMGraphHead> extends Serializable
PersistentVertexFactory<V extends EPGMVertex, E extends EPGMEdge> extends Serializable
PersistentEdgeFactory<E extends EPGMEdge, V extends EPGMVertex> extends Serializable
```

In `HBaseConfig`, it is not a restriction that GF VF and EF should extends factory, although they are named as `*Factory`.

Reply:
Hm.... In the original the definition was: GradoopStoreConfig<G extends EPGMGraphHead, V extends EPGMVertex, E extends EPGMEdge>

And G, V, E was used as generic parameter in the Persistent*Factorys. I'm not sure why it was done this way. 
Any thoughts on this @s1ck ? What do you think would be the best approach for this PR?

Reply:
@s1ck bump

Reply:
iirc, this changed a bit over time. At the moment, one could also use the interfaces directly instead of defining type parameters. Although I'm not sure of the implications, didn't touch that part of the code base for a while. Feel free to refactor as you see fit.

Reply:
ok, I guess this is not to important right now.